dist/extras/amd.js

```diff
@@ -3,132 +3,178 @@
+          dependencies[j] = dependencies[j]+".js";
         }
-      };
-    }];
-  }
+      }
+    }
 
-  // hook System.register to know the last declaration binding
-  let lastRegisterDeclare;
-  const systemRegister = systemPrototype.register;
-  // if we have named register support continue to use it
-  if (systemRegister.length === 3) {
-    systemPrototype.register = function (name, deps, declare) {
-      if (typeof name !== 'string')
-        lastRegisterDeclare = deps;
-      systemRegister.apply(this, arguments);
-    };
+    return [dependencies, depModules, setters];
   }
-  else {
-    systemPrototype.register = function (deps, declare) {
-      lastRegisterDeclare = declare;
-      systemRegister.apply(this, arguments);
-    };
-  }
-
-  const getRegister = systemPrototype.getRegister;
-  systemPrototype.getRegister = function () {
-    const register = getRegister.call(this);
-    // if its an actual System.register leave it
-    if (register && register[1] === lastRegisterDeclare)
-      return register;
 
-    // otherwise AMD takes priority
-    // no registration -> attempt AMD detection
-    if (!amdDefineDeps)
-      return register || emptyInstantiation;
+  /**
+   * Hook System.register
+   */
+  (function hookSystemRegister(lastRegisterDeclare) {
+    const systemRegister = systemPrototype.register;
+    // hook System.register to know the last declaration binding
+    // if we have named register support continue to use it
+    if (systemRegister.length === 3) {
+      systemPrototype.register = function (name, deps, declare) {
+        if (typeof name !== 'string')
+          lastRegisterDeclare = deps;
+        systemRegister.apply(this, arguments);
+      };
+    } else {
+      systemPrototype.register = function (deps, declare) {
+        lastRegisterDeclare = declare;
+        systemRegister.apply(this, arguments);
+      };
+    }
+  })(lastRegisterDeclare);
 
-    const registration = createAMDRegister(amdDefineDeps, amdDefineExec);
-    amdDefineDeps = null;
-    return registration;
-  };
-  let amdDefineDeps, amdDefineExec;
-  global.define = function (name, deps, execute) {
-    // define('', [], function () {})
-    if (typeof name === 'string') {
-      if (amdDefineDeps) {
-        if (!System.registerRegistry)
-          throw Error('Include the named register extension for SystemJS named AMD support.');
-        System.registerRegistry[name] = createAMDRegister(deps, execute);
-        amdDefineDeps = [];
-        amdDefineExec = emptyFn;
-        return;
+  /**
+   * Hook System.getRegister
+   */
+  (function hookGetRegister(amdDefineDependencies, amdDefineExecutor) {
+    const getRegister = systemPrototype.getRegister;
+    systemPrototype.getRegister = function () {
+      const register = getRegister.call(this);
+      // if its an actual System.register leave it
+      if (register && register[1] === lastRegisterDeclare) {
+        return register;
       }
-      else {
```

Comment:
Welp, this is actually part of global.define - moved to the top of the module to keep the logical order intact for people to understand how this module works